### PR TITLE
Fixed-Issue #4687:right panel to tab view in user page

### LINKED ISSUE #4687: Move right panel to tab view in user page
- [ ] Move the Right panel of the user page to tab view
- [ ] Redirect view all click to the appropriate tab in users page -> own data, follow data

### PR DESCRIPTION
**Pull request by harsh-vador**
### Describe your changes :

I worked on the right panel in user page which had user's data and following data that are now converted into a tab view because it makes it very much easy for a user to navigate between the tabs.
Closes #4687 

#
### Type of change :

- [x] Improvement

#
### Frontend Preview (Screenshots) :
Screenshot 2022-06-21 at 1 57 51 PM

#
https://user-images.githubusercontent.com/58542468/174755012-c1ecb177-f545-4a3d-8156-cbf3f349d5ed.mov

### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers

@open-metadata/ui

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/components/MyData/MyData.component.tsx [hunk 1]
```diff
@@ -134,10 +134,9 @@ const MyData: React.FC<MyDataProps> = ({
                 {ownedData.length ? (
                   <Link
                     data-testid="my-data"
-                    to={getExploreLinkByFilter(
-                      Ownership.OWNER,
-                      AppState.userDetails,
-                      AppState.nonSecureUserDetails
+                    to={getUserPath(
+                      AppState.getCurrentUserDetails()?.name || '',
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/MyData/MyData.component.tsx:138]: Can we store this value in a variable and use it everywhere?

```
const currentUser = AppState.getCurrentUserDetails();
```
```suggestion
                      currentUser?.name || '',
```

File: openmetadata-ui/src/main/resources/ui/src/components/Users/Users.component.test.tsx [hunk 2]
```diff
@@ -298,6 +287,19 @@ describe('Test User Component', () => {
...
 
+  it('Should check if cards are rendered', async () => {
+    const { container } = render(
+      <Users userData={mockUserData} {...mockProp} tab="mydata" />,
+      {
+        wrapper: MemoryRouter,
+      }
+    );
+
+    const obServerElement = await findByTestId(container, 'dataset-card');
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/Users/Users.component.test.tsx:298]:
```suggestion
    const datasetCardElement = await findByTestId(container, 'dataset-card');
```

**Review by Sachin-chaurasiya — CHANGES_REQUESTED** [review]
Few comments

**Comment by Sachin-chaurasiya** [comment]
@harsh-vador, Please revert the `package-lock.json` and `yarn-lock` changes.

File: openmetadata-ui/src/main/resources/ui/src/components/MyData/MyData.component.tsx (hunk 1 above)
Comment (harsh-vador) [reply openmetadata-ui/src/main/resources/ui/src/components/MyData/MyData.component.tsx:138]: Already tried this, wasn't working correctly

**Comment by vivekratnavel** [comment]
cc @devyani-kaushik @harshach 
Please check if the styling looks good for "My Data" and "Following" data

**Comment by vivekratnavel** [comment]
@harsh-vador We should not delete the yarn.lock file. Can you please bring it back?

**Review by ShaileshParmar11 — APPROVED** [review]
LGTM!

**Review by Sachin-chaurasiya — COMMENTED** [review]
Nit

**Review by Sachin-chaurasiya — APPROVED** [review]

**Pull request merged**